### ISSUE TITLE
AOT factory creation CLI command for `ReflectionBasedAbstractFactory` mapped factories

### ISSUE THREAD

**Issue by boesing** [laminas/laminas-servicemanager#171]
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | kinda
| BC Break    | no

#### Summary

Currently, there are CLI commands around where it is possible to create mezzio handler factories and stuff by using reflection. I was wondering if it would also be possible to provide a CLI command which takes runtime `config` from container and creates factories (along with a mapping-file such as `mezzio-tools`) using the same logic as `ReflectionBasedAbstractFactory`.

This would enable projects to use a more "generic" runtime factory during development while enabling devs to run that AoT CLI command in CI builds to auto-generate factories. This will remove the reflection stuff during runtime and increases performance with just running a CLI command (and maybe a bit of configuration, i.e. to provide different other PluginManager configuration keys such as `validators`, etc. - which can also be injected to the config from within our own components).

WDYT?

**Comment by boesing** [comment]
Seems that the `ReflectionBasedAbstractFactory` is supporting `config` parameter while the Factory creation tool is not. Might need to be extended to properly support the AOT command.

**Comment by froschdesign** [comment]
The documentation says:

> A parameter named $config typehinted as an array will receive the application "config" service (i.e., the merged configuration).

https://docs.laminas.dev/laminas-servicemanager/reflection-abstract-factory/

But should the entire configuration be used here? That just seems too much.

**Comment by boesing** [comment]
Thats not to be discussed in this issue.
This issue simply suggest providing an AOT CLI command which can be executed during CI which detects services explicitly using `ReflectionBasedAbstractFactory` as their factory.

The reflection based abstract factory provides that feature and thus the AOT command should reflect the same logic.
That it is not a good idea to use the whole config in a service in general is another topic.


So for example:

```php
return [
     'dependencies' => [
        'factories' => [
            'WhateverService' => ReflectionBasedAbstractFactory::class, // factory should be generated and configuration should be replaced with the generated factory
        ],
        'abstract_factories' => [
             ReflectionBasedAbstractFactory::class, // ignored
        ],
    ],
];
```

**Comment by boesing** [comment]
Implemented with #180
